### PR TITLE
Run one instead of n queries

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Instead of running the same query a bunch of times (e.g. for IN queries), this PR now runs a query with multiple OR clauses. This allows SQLite to dedupe results.

I think this should never hit a statement limit since our query complexity is quite limited.

### CODE REVIEW

**Comment by schmidt-sebastian** [comment]
cc @thebrianchen

**Review by wu-hui — COMMENTED** [review]
I want to see how you feel about an alternative: using SQL 'UNION' instead.

There are some advantages with UNION:
1. It is less sql text manipulating, the old code can be modified to simply concat the sql with UNION.
2. This would also be mentally close to IndexedDb and LevelDb.
3. My playing with sqlite query plan suggests UNION might be more efficient.. 

It looks like SQLite is not doing all optimization with connected OR filters, it will simply do a filter on `index_id`.

The query plan for using OR
```
sqlite> select * from index_entries_test where (index_id=1 and index_value >= 'x' and index_value <'z') or (index_id = 1 and index_value > 'a' and index_value <'c');
QUERY PLAN
`--SEARCH TABLE index_entries_test USING COVERING INDEX sqlite_autoindex_index_entries_test_1 (index_id=?)
```

The query plan for using UNION:
```
sqlite> select * from index_entries_test where (index_id = 1 and index_value > 'a' and index_value <'c')
   ...> union
   ...> select * from index_entries_test where (index_id=1 and index_value >= 'x' and index_value <'z')
   ...> order by index_value
   ...> limit 10
   ...> ;
QUERY PLAN
`--MERGE (UNION)
   |--LEFT
   |  `--SEARCH TABLE index_entries_test USING COVERING INDEX sqlite_autoindex_index_entries_test_1 (index_id=? AND index_value>? AND index_value<?)
   `--RIGHT
      `--SEARCH TABLE index_entries_test USING COVERING INDEX sqlite_autoindex_index_entries_test_1 (index_id=? AND index_value>? AND index_value<?)
```

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java [hunk 1]
```diff
@@ -176,12 +177,10 @@ public void addIndexEntries(Document document) {
     if (fieldIndex == null) return null;
 
     Bound lowerBound = target.getLowerBound(fieldIndex);
-    String lowerBoundOp = lowerBound.isBefore() ? ">=" : ">"; // `startAt()` versus `startAfter()`
-
     @Nullable Bound upperBound = target.getUpperBound(fieldIndex);
 
     if (Logger.isDebugEnabled()) {
-      Logger.warn(
+      Logger.debug(
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java:183]: It is meant to reduce computation complexity as it avoids stringifying the protos.

**Review by schmidt-sebastian — COMMENTED** [review]
Thanks for the great suggestion. Updated to use UNION, which is also a bit simpler.

File: firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (wu-hui) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteIndexManager.java:183]: Got it.

**Review by wu-hui — APPROVED** [review]

**Pull request merged**